ui/components/app/toast-master/toast-master.js

```diff
@@ -0,0 +1,299 @@
+  getUseNftDetection,
+} from '../../../selectors';
+import {
+  addPermittedAccount,
+  clearSwitchedNetworkDetails,
+  hidePermittedNetworkToast,
+} from '../../../store/actions';
+import {
+  AvatarAccount,
+  AvatarAccountSize,
+  AvatarNetwork,
+  Icon,
+  IconName,
+} from '../../component-library';
+import { Toast, ToastContainer } from '../../multichain';
+import { SurveyToast } from '../../ui/survey-toast';
+import {
+  selectNftDetectionEnablementToast,
+  selectShowConnectAccountToast,
+  selectShowPrivacyPolicyToast,
+  selectShowSurveyToast,
+  selectSwitchedNetworkNeverShowMessage,
+} from './selectors';
+import {
+  setNewPrivacyPolicyToastClickedOrClosed,
+  setNewPrivacyPolicyToastShownDate,
+  setShowNftDetectionEnablementToast,
+  setSurveyLinkLastClickedOrClosed,
+  setSwitchedNetworkNeverShowMessage,
+} from './utils';
+
+export function ToastMaster() {
+  const location = useLocation();
+
+  const onHomeScreen = location.pathname === DEFAULT_ROUTE;
+
+  return (
+    onHomeScreen && (
+      <ToastContainer>
+        <SurveyToast />
+        <ConnectAccountToast />
+        <SurveyToastMayDelete />
+        <PrivacyPolicyToast />
+        <SwitchedNetworkToast />
+        <NftEnablementToast />
+        <PermittedNetworkToast />
+      </ToastContainer>
+    )
+  );
+}
+
+function ConnectAccountToast() {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+
+  const [hideConnectAccountToast, setHideConnectAccountToast] = useState(false);
+  const account = useSelector(getSelectedAccount);
+
+  // If the account has changed, allow the connect account toast again
+  const prevAccountAddress = usePrevious(account?.address);
+  if (account?.address !== prevAccountAddress && hideConnectAccountToast) {
+    setHideConnectAccountToast(false);
+  }
+
+  const showConnectAccountToast = useSelector((state) =>
+    selectShowConnectAccountToast(state, account),
+  );
+
+  const activeTabOrigin = useSelector(getOriginOfCurrentTab);
+
+  return (
+    Boolean(!hideConnectAccountToast && showConnectAccountToast) && (
+      <Toast
+        dataTestId="connect-account-toast"
+        key="connect-account-toast"
+        startAdornment={
+          <AvatarAccount
+            address={account.address}
+            size={AvatarAccountSize.Md}
+            borderColor={BorderColor.transparent}
+          />
+        }
+        text={t('accountIsntConnectedToastText', [
+          account?.metadata?.name,
+          getURLHost(activeTabOrigin),
+        ])}
+        actionText={t('connectAccount')}
+        onActionClick={() => {
+          // Connect this account
+          dispatch(addPermittedAccount(activeTabOrigin, account.address));
+          // Use setTimeout to prevent React re-render from
+          // hiding the tooltip
+          setTimeout(() => {
+            // Trigger a mouseenter on the header's connection icon
+            // to display the informative connection tooltip
+            document
```

Comment:
Seems odd that we have to interact with `document` directly here. More so by using a CSS selector tied to a data-testids (which I think we should constrain to tests). Can we implement this w/o direct interaction with the DOM?

Reply:
This is something that used to be in `routes.component.js` and I just left it alone.  Do you think we should pass an event that changes the `open` prop of the `Tooltip`?